tests/queries/0_stateless/01233_check_part_meta_cache_replicated_in_atomic.sql

```diff
@@ -0,0 +1,128 @@
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+--Delete some data.
+alter table test_metadata_cache.check_part_metadata_cache delete where k = 1;
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+alter table test_metadata_cache.check_part_metadata_cache delete where k = 8;
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+-- Delete some data.
+INSERT INTO test_metadata_cache.check_part_metadata_cache (p, k, v1, v2) VALUES ('2018-06-15', 5, 1000, 2000), ('2018-06-16', 6, 3000, 4000), ('2018-06-17', 7, 5000, 6000), ('2018-06-18', 8, 7000, 8000);
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+-- Drop partitioin 201805
+alter table test_metadata_cache.check_part_metadata_cache drop partition 201805;
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+-- Optimize table.
+INSERT INTO test_metadata_cache.check_part_metadata_cache (p, k, v1, v2) VALUES ('2018-06-15', 5, 1000, 2000), ('2018-06-16', 6, 3000, 4000), ('2018-06-17', 7, 5000, 6000), ('2018-06-18', 8, 7000, 8000);
+optimize table test_metadata_cache.check_part_metadata_cache FINAL;
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+-- Add column.
+alter table test_metadata_cache.check_part_metadata_cache  add column v3 UInt64;
+INSERT INTO test_metadata_cache.check_part_metadata_cache (p, k, v1, v2) VALUES ('2018-06-15', 5, 1000, 2000), ('2018-06-16', 6, 3000, 4000), ('2018-06-17', 7, 5000, 6000), ('2018-06-18', 8, 7000, 8000);
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+-- Add TTL info.
+alter table test_metadata_cache.check_part_metadata_cache  modify TTL p + INTERVAL 10 YEAR;
+INSERT INTO test_metadata_cache.check_part_metadata_cache (p, k, v1, v2) VALUES ('2018-06-15', 5, 1000, 2000), ('2018-06-16', 6, 3000, 4000), ('2018-06-17', 7, 5000, 6000), ('2018-06-18', 8, 7000, 8000);
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
+
+-- Modify TTL info.
+alter table test_metadata_cache.check_part_metadata_cache  modify TTL p + INTERVAL 15 YEAR;
+INSERT INTO test_metadata_cache.check_part_metadata_cache (p, k, v1, v2) VALUES ('2018-06-15', 5, 1000, 2000), ('2018-06-16', 6, 3000, 4000), ('2018-06-17', 7, 5000, 6000), ('2018-06-18', 8, 7000, 8000);
+with arrayJoin(checkPartMetadataCache('test_metadata_cache', 'check_part_metadata_cache')) as info select countIf(info.5 = 0);
```

Comment:
Worth adding a test with `MOVE [PARTITION|PART] TO [DISK|VOLUME]` ?

Reply:
ok .

Reply:
done. See  tests/integration/test_multiple_disks/test.py